### ISSUE TITLE
Improve error message when PKG_CONFIG_PATH is misconfigured

### ISSUE THREAD

**Issue by lappkast** [EventStore/EventStore#464]
Hi,

I'm trying to build c6fd6391 and I'm ending up with this message from build.sh.
 
Package monosgen-2 was not found in the pkg-config search path.
Perhaps you should add the directory containing `monosgen-2.pc'
to the PKG_CONFIG_PATH environment variable
No package 'monosgen-2' found
Packaging requires monosgen-2 to be discoverable via pkg-config. Aborting.

**Comment by gregoryyoung** [comment]
I get the same here. This was added to "help" people in case they didn't
have things setup properly to build my guess is its not testing the right
thing. cc @james

I am removing it for now. We can track re-adding it.

Done here:
https://github.com/EventStore/EventStore/commit/457cc2ba39833f6dc853b0af0fa6787cb351fa4f

Greg

On Fri, May 1, 2015 at 1:01 PM, Mikko Lappalainen <notifications@github.com>
wrote:

> Hi,
>
> I'm trying to build c6fd639
> <https://github.com/EventStore/EventStore/commit/c6fd6391716d52f5ed9eb854e03b83946a5b1c6c>
> and I'm ending up with this message from build.sh.
>
> Package monosgen-2 was not found in the pkg-config search path.
> Perhaps you should add the directory containing `monosgen-2.pc'
> to the PKG_CONFIG_PATH environment variable
> No package 'monosgen-2' found
> Packaging requires monosgen-2 to be discoverable via pkg-config. Aborting.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EventStore/EventStore/issues/464>.
>



-- 
Studying for the Turing test

**Comment by haf** [comment]
This change in mono is probably this one:

```
commit 455debc3d4f9edb184ec8eb70cd6b4eb2d0d1519
Author: Alex Rønne Petersen <alexrp@xamarin.com>
Date:   Tue Apr 7 14:28:46 2015 +0200

    Remove spec file as we don't use it anymore.

    See: https://github.com/directhex/mono-snapshot
```

Which removes `mono-core.spec.in` completely in favor of https://github.com/directhex/mono-snapshot

**Comment by lappkast** [comment]
Awsome

**Comment by jen20** [comment]
I don't think this is the issue. My installation of 4.0.1 from built for x64 from source still contains the package files, it's likely that you actually just don't have `$PKG_CONFIG_PATH` set to point to your mono installation.

I'll improve the part of the error message we produce - "Perhaps you should add the directory containing `monosgen-2.pc' to the PKG_CONFIG_PATH environment variable" comes directly from pkgconfig.

**Comment by haf** [comment]
I verified this is the issue and that it works with Greg's fix. ES compiles and runs fine with the latest installation. It's a 32 bit installation however, as that's the default. It's just for dev though.

**Comment by gregoryyoung** [comment]
if you set the environment variable it should work as well. I just removed the check its not actually needed to build only to package. Setting the environment variable properly is a better answer

**Comment by haf** [comment]
Yes, that's a better improvement, I agree. Document in README?

**Comment by gregoryyoung** [comment]
Just putting better message on the error failure in case you get it. This
is something that should already be setup with mono

On Fri, May 1, 2015 at 5:26 PM, Henrik Feldt <notifications@github.com>
wrote:

> Yes, that's a better improvement, I agree. Document in README?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EventStore/EventStore/issues/464#issuecomment-98144377>
> .
>



-- 
Studying for the Turing test

**Comment by jen20** [comment]
It already is in the README: "Mono version 3.2.3 (minimum version, preferably later) development packages on `PATH`, and registered with `pkg-config`, such that `which xbuild` and `pkg-config --cflags monosgen-2` return a zero return codes."

Could use the grammar fixing up though.

**Comment by haf** [comment]
What should the library path be on OS X?

**Comment by jen20** [comment]
`/path/to/mono/lib/pkgconfig​`. In my case:

```
$ echo $PKG_CONFIG_PATH
/opt/monos/current/lib/pkgconfig
```

**Comment by haf** [comment]
I see, thanks. By default on OS X, it's `/Library/Frameworks/Mono.framework/Libraries/pkgconfig/`

**Comment by jen20** [comment]
Ah, possibly. I'd have expected the Xamarin installer to have configured
that correctly, though I'm not entirely sure why I would have expected
that.​ I'll edit the README include a section about using the Xamarin
packages.

**Comment by haf** [comment]
You might mention they are 32 bit by default, too, and that ES normally wants 64 bits.

**Comment by jen20** [comment]
Yes, that's why we recommend building from source. I have no idea why 32
bit packages even still exist in 2015.

**Comment by jen20** [comment]
Fixed in 8c9d824a1d0e29d1720359b25a2874b916242731.
